src/map/skill.cpp

```diff
@@ -2441,12 +2441,11 @@ int skill_counter_additional_effect (struct block_list* src, struct block_list *
 		sc_start(src,src,SC_BLIND,2*skill_lv,skill_lv,skill_get_time2(skill_id,skill_lv));
 		break;
 	case HFLI_SBR44:	//[orn]
-	case HVAN_EXPLOSION:
 		if(src->type == BL_HOM){
-			TBL_HOM *hd = (TBL_HOM*)src;
-			hd->homunculus.intimacy = (skill_id == HFLI_SBR44) ? 200 : 100; // hom_intimacy_grade2intimacy(HOMGRADE_HATE_WITH_PASSION)
+			TBL_HOM *hd = BL_CAST(BL_HOM,src);
```

Comment:
```suggestion
			struct homun_data *hd = (struct homun_data *)src;
```
Please use normal cast here.
The BL type was already checked above.